### DISCUSSION TITLE
Scheduled process restart

### DISCUSSION THREAD

**Discussion by Leokuma** [Hexagon/pup#18] [Ideas]
Pm2 has this option `cron_restart` which takes a cron pattern to restart your process.

At first I thought we could do the same, but then I thought: what if we had a `terminate` option instead? I think it would be more flexible.

Config     | Behavior | 
---------- | ---------
autostart: true,<br>terminate: 0 0 * * * | Restart at midnight
autostart: false,<br>cron: 0 6 * * *,<br>terminate: 0 22 * * * | Run everyday from 6am to 10pm
autostart: false,<br>cron: * * * * 5,<br> terminate: * * * * 6 | Run on Fridays only

Maybe the options names should be changed in that case.

That would allow manually scaling your processes. For example, if you know you have more users on friday, you could schedule an extra cluster for fridays.

Another use case for example would be an email service that you don't want to run late at night.

Cron restart would be useful for me because I have broken, legacy programs that need to be restarted everyday.

Not sure how `terminate` would work with `overrun`, though.

**Comment by Hexagon** [comment]
Very interesting, the terminte option seem more flexible than a cron_restart feature 👍 

Cron-terminate wouldn't interfere with overrun. Overrun only block starts when a process is already running.

**Comment by Hexagon** [comment]
Implemented in [1.0.0-beta.13](https://github.com/Hexagon/pup/releases/tag/1.0.0-beta.13)

**Reply by Leokuma** [reply]
Oh that was fast! Thank you!
